src/toil/leader.py

```diff
@@ -260,7 +260,12 @@ def innerLoop(self):
                             continue
 
                         # If the job is a checkpoint and has remaining retries then reissue it.
-                        elif jobGraph.checkpoint is not None and jobGraph.remainingRetryCount > 0:
+                        # The logic behind using > 1 rather than > 0 here: Since this job has
+                        # been tried once (without decreasing its retry count as the job
+                        # itself was successful), and its subtree failed, it shouldn't be retried
+                        # unless it has more than 1 try.
+                        elif jobGraph.checkpoint is not None and jobGraph.remainingRetryCount > 1:
```

Comment:
OK, @cket, @benedictpaten, I now remember why this is correct. The "remainingRetryCount" is set to AbstractJobStore._defaultTryCount() [in all the jobStores (e.g. fileJobStore)](https://github.com/BD2KGenomics/toil/blob/b27a8efadb82089a2799415d8a506bd61758aead/src/toil/jobStores/fileJobStore.py#L96). Then [it's passed as `tryCount` to the jobGraph](https://github.com/BD2KGenomics/toil/blob/b27a8efadb82089a2799415d8a506bd61758aead/src/toil/jobGraph.py#L122). And indeed it is a try count: [_defaultTryCount is set to retryCount + 1](https://github.com/BD2KGenomics/toil/blob/b27a8efadb82089a2799415d8a506bd61758aead/src/toil/jobStores/abstractJobStore.py#L950).

I'd vote to rename it remainingTryCount.

Maybe some of the other job types are tried one too many times as well?